### PR TITLE
Use moditect for multi-release module-info

### PR DESCRIPTION
**Pull request by Meodinger**
Jackson was using [moditect](https://github.com/moditect/moditect) to generate multi-release module. 
I've tested on release 1.14.3.

### CODE REVIEW

**Comment by Meodinger** [comment]
Strange, github checks all failed because of moditect: Invocation of jdeps failed

**Comment by Meodinger** [comment]
Tested on master, `jsr305` is not required in module-info because it's in the scope `provided` and it's not modular.

File: pom.xml [hunk 1]
```diff
@@ -123,9 +131,6 @@
         <version>3.2.0</version>
         <configuration>
           <archive>
-            <manifestEntries>
-              <Automatic-Module-Name>org.jsoup</Automatic-Module-Name>
```
Comment (jhy) [inline pom.xml:127]: Where does the module name come from now - is it line 247 module artifact?

File: src/main/moditect/module-info.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module org.jsoup {
+    requires static java.xml;
+
+    exports org.jsoup;
```
Comment (jhy) [inline src/main/moditect/module-info.java:4]: Don't we need to export .helper as well? I am concerned that tests didn't fail for this, are there tests to ensure we are exporting the correct packages?

Comment (Meodinger) [reply src/main/moditect/module-info.java:4]: Actually, I'm not very clear with the package structure of Jsoup and not very sure who should be exported and who should not, so I just export `org.jsoup`. I'll add the necessary `exports` if you suggest some.

**Comment by jhy** [comment]
Thanks all - I have module support with #2025.

It would be great if folks could test this out and report back with confirmation of success or any issues. Please report back on the PR #2025 (or a new issue if any problems) as there are a few duplicate issues for this. To use a snapshot, `git pull && mvn install`, and use version `1.17.1-SNAPSHOT` in your project.

**Pull request closed without merging**